core/src/components/progress-bar/progress-bar.tsx

```diff
@@ -83,13 +83,25 @@ const renderIndeterminate = () => {
   ];
 };
 
-const renderProgress = (value: number, buffer: number) => {
+const renderProgress = (value: number, buffer: number, reversed = false) => {
   const finalValue = clamp(0, value, 1);
   const finalBuffer = clamp(0, buffer, 1);
+  const dirScale = reversed ? -1 : 1;
 
   return [
     <div class="progress" style={{ transform: `scaleX(${finalValue})` }}></div>,
-    finalBuffer !== 1 && <div class="buffer-circles"></div>,
-    <div class="progress-buffer-bar" style={{ transform: `scaleX(${finalBuffer})` }}></div>,
+    <div
+      class="buffer-circles-wrapper"
+      style={{ left: `${finalBuffer * 100 * dirScale}%` }}
```

Comment:
Is there a reason why we are using `left` rather than a `translateX`?

Reply:
It's because transform is already used for the animation of the elemen `.buffer-circles`. If you want to use transform you have to wrap it again in another element. So that you can move the wrapper and have still the transform animation to the buffer dotted element. Or vice versa. (use `left` for the animation)

https://github.com/ionic-team/ionic-framework/blob/5ed73cdf4d63eeee25ef28d9676fcaa4f8e07b47/core/src/components/progress-bar/progress-bar.scss#L260-L264

Some background of this solution. It works because the dotted buffer animation (`.buffer-circles`) is shifting behind the filled buffer element (`.progress-buffer-bar`). So that it's not under the semi-transparent `.progress-buffer-bar` element. (The initial issue.)
I wrapped the `.buffer-circles` element to use overflow hidden (which requires the `position` attribute set.) Then moving the wrapper (`.buffer-circles-wrapper`) after the buffer-bar. 
But this movement would affect the animation. The dots animation moves from `0` to `-10px` using transform. And if you increase the buffer progress, the dots (aka circles) appears to be standing still or moving in another direction and jiggling.
For this reason the dots are moving back inside the wrapper. Synced by the movement of the buffer progress bar and transition. ... This is maybe not the best solution I ever made. But works. I tested it on Windows 10 latest Chrome and iPhone SE 2 iOS 14.2 Safari.